### ISSUE TITLE
relate environmental variables to OTU clusters in network

### ISSUE THREAD

**Issue by Biyorun** [joey711/phyloseq#473]
Hi,

how could I correlate OTU clusters in a network to environmental parameters? Is this possible using the sample_data file in phyloseq? 
Are there other possibilities? How do you people do this?

Thanks!

**Comment by Biyorun** [comment]
Anybody?

**Comment by spholmes** [comment]
This is a statistical methodology question not  a phyloseq question,
the standard methods people are currently using involve the vegan
function adonis for instance, you might look at some of the online workflow
examples
provided by Joey:
http://joey711.github.io/phyloseq-demo/Restroom-Biogeography
also see
http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3357092/

Best
Susan

On Fri, May 8, 2015 at 2:59 AM, Biyorun <notifications@github.com> wrote:

> Hi,
>
> how could I correlate OTU clusters in a network to environmental
> parameters? Is this possible using the sample_data file in phyloseq?
> Are there other possibilities? How do you people do this?
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joey711/phyloseq/issues/473>.
>



-- 
Susan Holmes
Professor, Statistics and BioX
Director, MCS
Sequoia Hall,
390 Serra Mall
Stanford, CA 94305
http://www-stat.stanford.edu/~susan/

**Comment by joey711** [comment]
I agree with Susan. However, this is a much-requested feature. Any ideas for smoothing or better documenting this? I wouldn't want to imply that my Restroom example covers all needs.

**Comment by audy** [comment]
For comparing one set of clusterings versus another label, you can use the homogeneity and completes scores, or the harmonic mean of the two (v_measure).

There is some documentation on these in the SciKit-Learn package in Python: http://scikit-learn.org/stable/modules/generated/sklearn.metrics.homogeneity_completeness_v_measure.html#sklearn.metrics.homogeneity_completeness_v_measure. There is probably an implementation in R.

**Comment by joey711** [comment]
Thanks @audy 

Between
http://cran.r-project.org/web/views/Cluster.html ,
http://cran.r-project.org/web/views/gR.html , 
and
http://cran.r-project.org/web/views/MachineLearning.html
(among many others), I think R has this covered. 

The OP, @Biyorun , has not made clear precisely what kind of test is desired. What network? Are the nodes "clusters of OTUs", or "OTU clusters" (AKA, OTUs)? What is the hypothesis test or general nature of the inference that needs to be tested.

I think Susan's answer satisfies this issue, so I will close for now. However, please feel free to post back with reproducible example code and/or links to tutorials or blog posts where a relevant example is demonstrated.

Cheers

joey

**Comment by Biyorun** [comment]
Dear Dr. McMurdie,

it are indeed clustered OTUs in a network. (see figure), not sequences 
clustered into OTUs.

I would like to superimpose/add the environmental variables that 
correlate/are significantly associated with the observed clusters. 
Basically my question was if this can be done directly from the physeq 
object (with OTU and ENV files), or if this has (can?) to be done in 
another way?

Furthermore, is it possible to use unweighted lines in the plot_net 
function?


Thanks in advance.

Kind regards,
Bjorn



Paul J. McMurdie schreef op 19/05/2015 om 23:37:
